pkg/gossiper/db.go

```diff
@@ -19,31 +20,31 @@ func Newdb(path string) (int, *mydb) {
 	newdb := &mydb{}
 	newdb.path = path
 	var err error
-	newdb.db, err = leveldb.OpenFile("path/to/db", nil)
+	newdb.db, err = leveldb.OpenFile(newdb.path, nil)
```

Comment:
You want to return the error. If you want the error code, that is fine. You can return that too. But you have to return the error as well for the `newDB` Func.